mmstudio/src/main/java/org/micromanager/acquisition/internal/acqengjcompat/AcqEngJAdapter.java

```diff
@@ -448,14 +417,19 @@ private Iterator<AcquisitionEvent> createAcqEventIterator(SequenceSettings acqui
          zStack = MDAAcqEventModules.zStack(0,
                acquisitionSettings.slices().size() - 1,
                acquisitionSettings.sliceZStepUm(),
-               origin);
+               origin,
+               chSpecs);
       }
 
       if (acquisitionSettings.useChannels()) {
-         channels = MDAAcqEventModules.channels(acquisitionSettings.channels());
-         //TODO: keep shutter open
-         //TODO: skip frames
-         //TODO: z stack off for channel
+         if (chSpecs.size() > 0) {
+            Integer middleSliceIndex = (acquisitionSettings.slices().size() - 1) / 2;
+            channels = MDAAcqEventModules.channels(chSpecs, middleSliceIndex);
+         }
+         // TODO: keep shutter open, I believe this needs to be done in AcqEngJ; there should be
```

Comment:
I think the way to implement the shutter behavior is with an acquisition hook, because we're trying to move towards not hard coding particular microscope assumptions in the acqEng and this shutter behvaior is one of them

Reply:
Seems to me that shutter behavior is super similar to sequencing.  Also, controlling light exposure to the sample seems like a very common (universal?) part of microscope acquisitions.  

I guess this is doable as a Hook, I can give a try after getting the Autofocus working.

Reply:
For most cases yes, though its much more of a concern for fluorescence than other modalities. But I'm not sure I would call it universal. I agree that it is universal for using MMStudio, and thus my feeling is its best to have it in that layer, to leave open the possibility of people using AcqEngJ in unexpected ways

Reply:
Implemented in an acquisition Hook.